_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,201 @@
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 2
+
+The `<svg>` element has a role of `img` and an `aria-label` containing the accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 3
+
+The `<svg>` element has the role of `graphics-document` which takes its accessible name from its' child content, the `<title>` and `<desc>` elements.
+
+```html
+
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document">
+    <title>3 circles</title>
+    <desc>3 yellow circles with a green stroke horizontally adjacent to each other
+    </desc>
+    <g role="group">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+    </g>
+</svg>
+```
+
+#### Pass example 4
+
+The `<circle>` element is included in the accessibility tree and takes its' accessible name from its' child content, the `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg">
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow">
+	    <title>Yellow circle</title>
+	</circle>
+</svg>
+```
+
+#### Pass example 5
+
+The `svg` element and both child `<g>` elements are included in the accessibility tree with the `graphics-document` and `graphics-object` roles. All three elements have accessibile names via the `title` and `aria-label` attributes.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="graphics-document">
+	<title>3 shapes</title>
+	<g role="graphics-object" aria-label="Blue rectangle" transform="translate(100,100)">
+        <desc>Large blue rectangle with two other shapes inside it</desc>
+		<rect fill="blue" role="graphics-object" aria-label=" large blue rectangle" width="200" height="100" y="-100" />
+		<g role="graphics-object" aria-label="small yellow rectangle in larger green rectangle" transform="translate(30,-90)">
+			<rect fill="green" width="50" height="90"/>
+			<rect fill="yellow" x="5" y="5" width="40" height="30" />
+			</g>
+		 </g>
+	</svg>
+```
+
+### Failed
+
+#### Failed example 1
+
+The `<svg>` element has a role of `img` but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `img` role includes the `<svg>` element in the accessibility tree but the element has an empty accessible name `title` attribute.
```

Comment:
Yes, but you expectation is "Each target element has an accessible name.", not "each target has an accessible name that is relevant",  nor "has an accessible name that is not empty" ? I think is different.

In the "image has a name" rule, you write : 

> Each target element has an [accessible name](#accessible-name) that is [non-empty](#non-empty) or is marked as [decorative](#decorative).

Well there's a lot of discussion on the "non empty" term, but in the "heading is descriptive" rule it is written with new terms : "has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace)."

I think the expectation needs to be a little more specific when I take a look to other rules. 

>  Each target element has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace).

Reply:
I agree with @audreymaniez in that we should make it explicit that the accessible name is not only whitespace. Otherwise, I'm not sure if this is an inapplicable or not. If the title is not empty then it will be included in the accessibility tree. But if the title is empty after trimming whitespace then the spec says "User agents MAY include elements with these child elements without checking for valid text content." 
If it's up to the UA to decide whether to include the element in the accessibility tree I don't think it is a good idea to leave this open in the applicability. Following @audreymaniez suggestion would take care of this issue.

Reply:
I support the suggestions from @audreymaniez and @carlosapaduarte . Also, the example description should read "title element", not "title attribute".